app/src/main/java/io/homeassistant/companion/android/sensors/SensorsSettingsFragment.kt

```diff
@@ -20,28 +22,73 @@ class SensorsSettingsFragment : PreferenceFragmentCompat() {
     private val refresh = object : Runnable {
         override fun run() {
             SensorWorker.start(requireContext())
+            totalDisabledSensors = 0
+            totalEnabledSensors = 0
             val sensorDao = AppDatabase.getInstance(requireContext()).sensorDao()
-            SensorReceiver.MANAGERS.forEach { managers ->
-                managers.availableSensors.forEach { basicSensor ->
+            SensorReceiver.MANAGERS.sortedBy { it.name }.filter { it.hasSensor(requireContext()) }.forEach { managers ->
+                managers.availableSensors.sortedBy { it.name }.forEach { basicSensor ->
                     findPreference<Preference>(basicSensor.id)?.let {
                         val sensorEntity = sensorDao.get(basicSensor.id)
+                        val sensorSettings = sensorDao.getSettings(basicSensor.id)
+                        if (!sensorSettings.isNullOrEmpty()) {
+                            sensorSettings.forEach { setting ->
+                                if (!setting.name.isNullOrBlank())
+                                    if (getString(basicSensor.name) !in sensorsWithSettings) {
+                                        sensorsWithSettings += getString(basicSensor.name)
+                                        sensorsWithSettings.sort()
+                                    }
+                            }
+                        }
                         if (sensorEntity?.enabled == true) {
+                            totalEnabledSensors += 1
                             if (basicSensor.unitOfMeasurement.isNullOrBlank())
                                 it.summary = sensorEntity.state
                             else
                                 it.summary = sensorEntity.state + " " + basicSensor.unitOfMeasurement
                             // TODO: Add the icon from mdi:icon?
                         } else {
+                            totalDisabledSensors += 1
                             it.summary = "Disabled"
                         }
                     }
                 }
             }
+
+            findPreference<PreferenceCategory>("enable_disable_category")?.let {
+                it.summary = getString(R.string.manage_all_sensors_summary, (totalDisabledSensors + totalEnabledSensors))
+            }
+
+            findPreference<SwitchPreference>("enable_disable_sensors")?.let {
+                if (totalDisabledSensors == 0) {
+                    it.title = getString(R.string.disable_all_sensors, totalEnabledSensors)
+                    it.summary = ""
+                    it.isChecked = true
+                } else {
+                    if ((totalDisabledSensors + totalEnabledSensors) == totalDisabledSensors)
```

Comment:
Isn't this just the following?
```suggestion
                    if (totalEnabledSensors == 0)
```